app/services/friends_service.rb

```diff
@@ -0,0 +1,21 @@
+class FriendsService
+  class << self
+    def invite(emails)
+      @incorrect_emails = ''
+      emails.delete_if do |email|
+        if email[/\A([a-z0-9_\.-]+)@([\da-z\.-]+)\.([a-z\.]{2,6})\z/i]
+          User.exists?(email: email)
+        else
+          @incorrect_emails += if @incorrect_emails.blank?
+                                 email
+                               else
+                                 ', ' + email
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.